cpp/include/raft/matrix/detail/select_k-ext.cuh

```diff
@@ -44,6 +45,16 @@ void select_k(raft::resources const& handle,
               bool sorted       = false,
               SelectAlgo algo   = SelectAlgo::kAuto,
               const IdxT* len_i = nullptr) RAFT_EXPLICIT;
+
+template <typename T, typename IdxT>
+void select_k(raft::resources const& handle,
```

Comment:
Sorry, @rhdong, I do have a nitpick here, now that I finally have time to go through this PR. I'm okay doing this change in an immediate follow-on, but the sparse k-selection should be in raft/sparse/matrix, not in the dense APIs.

Reply:
Hi @cjnolet , thank you for the comment! I just moved the public API to the `raft::sparse` scope. Considering many shared source codes, I temporarily didn't move more code to the sparse directory like `detail::select_k`; if you prefer more, please let me know.

Reply:
Sorry @rhdong please move the impl details over to sparse also. This is introducing too much confusion having them not be co-located.

Reply:
> Sorry @rhdong please move the impl details over to sparse also. This is introducing too much confusion having them not be co-located.

Sure, I will